br/cmd/br/cmd.go

```diff
@@ -107,6 +110,23 @@ func AddFlags(cmd *cobra.Command) {
 	_ = cmd.PersistentFlags().MarkHidden(FlagRedactLog)
 }
 
+func calculateMemoryLimit(memleft uint64) uint64 {
+	const halfGB = uint64(512 * 1024 * 1024)
+	const fourGB = 8 * halfGB
+	// memreserved = f(memleft) = 512MB * memleft / (memleft + 4GB)
+	//  * f(0) = 0
+	//  * f(4GB) = 256MB
+	//  * f(+inf) -> 512MB
+	memreserved := halfGB / (1 + fourGB/(memleft+1))
```

Comment:
extreme case, if memleft is unit64 max it's going to overflow I guess

Reply:
in what case would memleft be uint64max (= 16 EiB) 👀 

anyway changing `memleft + 1` to `memleft | 1` should be good enough

Reply:
in the unexpected error case I would say, the `memleft` is passed down by  `memleft := memtotal - memused`, if `memtotal < memused`, it can underflow as in my other comment. I feel like we probably should explicitly check and make sure it won't go wrong in any extreme case, cuz a lot of assumptions are false when running in production.

Reply:
done. And add check `memtotal < memused` in another comment.